app/src/main/java/org/sil/gatherwords/MainActivity.java

```diff
@@ -32,10 +35,14 @@
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+    }
 
+    @Override
+    protected void onResume() {
+        super.onResume();
+        // Moved to onResume() so that the ListView is reconstructed when the back button was pressed
```

Comment:
Must all of this code be repeated each `onResume()`?  Visually it seems all you would need here is `FillSessionListTask()`.

Reply:
The answer is no, they do not. Updated.